src/pyuvsim/simsetup.py

```diff
@@ -1911,129 +2012,109 @@ def initialize_uvdata_from_params(
+        uvparam_dict["antenna_positions"] = tele_params["antenna_positions"]
+        if "x_orientation" in tele_params:
+            uvparam_dict["x_orientation"] = tele_params["x_orientation"]
+
+        uvparam_dict["ellipsoid"] = tele_params["ellipsoid"]
+
+        uv_obj = UVData.new(
+            telescope_location=telescope_location,
+            phase_center_catalog=phase_center_catalog,
+            vis_units="Jy",
+            history="",
+            do_blt_outer=True,
+            time_axis_faster_than_bls=True,
+            **uvparam_dict,
+        )
+        uv_obj.telescope_location = np.asarray(uv_obj.telescope_location)
 
-    uv_obj.set_lsts_from_time_array()
-    # set the apparent coordinates on the object
-    uv_obj._set_app_coords_helper()
+        # This has to be changed to avoid a bug in uvfits for 1 freq channel.
+        # The bug is fixed in v3.0
+        uv_obj.flex_spw = False
+        uv_obj._flex_spw_id_array.required = False
 
-    # add other required metadata to allow select to work without errors
-    # these will all be overwritten in uvsim._complete_uvdata, so it's ok to hardcode them here
-    logger.info("Set Metadata")
     logger.info(f"  Baseline Array: {uv_obj.baseline_array.nbytes / 1024**3:.2f} GB")
     logger.info(f"      Time Array: {uv_obj.time_array.nbytes / 1024**3:.2f} GB")
     logger.info(f"Integration Time: {uv_obj.integration_time.nbytes / 1024**3:.2f} GB")
     logger.info(f"       Ant1Array: {uv_obj.ant_1_array.nbytes / 1024**3:.2f} GB")
-
-    uv_obj.set_lsts_from_time_array()
-    uv_obj.set_uvws_from_antenna_positions()
     logger.info(f"BLT-ORDER: {uv_obj.blt_order}")
-    logger.info("Set UVWs")
-
-    uv_obj.history = ""
-
-    uv_obj.vis_units = "Jy"
-
-    uv_obj.instrument = uv_obj.telescope_name
-
-    uv_obj.spw_array = np.array([0])
-    uv_obj.flex_spw_id_array = np.full(uv_obj.Nfreqs, uv_obj.spw_array[0], dtype=int)
-    if uv_obj.Ntimes == 1:
-        uv_obj.integration_time = np.ones_like(
-            uv_obj.time_array, dtype=np.float64
-        )  # Second
-    else:
-        # Note: currently only support a constant spacing of times
-        uv_obj.integration_time = (
-            np.ones_like(uv_obj.time_array, dtype=np.float64)
-            * np.diff(np.unique(uv_obj.time_array))[0]
-            * (24.0 * 60**2)  # Seconds
-        )
+    logger.info("Initialized UVData object")
 
     subselect(uv_obj, param_dict)
     logger.info("After Select")
 
-    # we construct uvdata objects in (time, ant1) order
-    # but the simulator will force (time, baseline) later
-    # so order this now so we don't get any warnings.
-    if reorder_blt_kw is None:
-        reorder_blt_kw = {"order": "time", "minor_order": "baseline"}
-
-    if reorder_blt_kw and reorder_blt_kw != {
-        "order": uv_obj.blt_order[0],
-        "minor_order": uv_obj.blt_order[1],
-    }:
-        uv_obj.reorder_blts(**reorder_blt_kw)
+    ordering(uv_obj, param_dict, reorder_blt_kw)
```

Comment:
this feels weird... can the name of the function be a verb like `reorder_blts` or `set_order`?

Reply:
sure.